components/claim-mgt/org.wso2.carbon.identity.claim.metadata.mgt/src/main/java/org/wso2/carbon/identity/claim/metadata/mgt/ClaimMetadataManagementServiceImpl.java

```diff
@@ -648,6 +652,95 @@ public List<Claim> getMappedExternalClaimsForLocalClaim(String localClaimURI, St
+            if (!entry.getKey().startsWith(ClaimConstants.PROFILES_CLAIM_PROPERTY_PREFIX)) {
+                continue;
+            }
+
+           String[] profilePropertyKeyArray = entry.getKey().split("\\.");
+           if (profilePropertyKeyArray.length < 2 || !allowedClaimProfiles.contains(profilePropertyKeyArray[1])) {
+               throw new ClaimMetadataClientException(ERROR_CODE_INVALID_ATTRIBUTE_PROFILE);
+           }
+        }
+
+        String[] allowedProfilePropertyKeys = {
+                ClaimConstants.SUPPORTED_BY_DEFAULT_PROPERTY,
+                ClaimConstants.REQUIRED_PROPERTY,
+                ClaimConstants.READ_ONLY_PROPERTY
+        };
+        for (String propertyKey: allowedProfilePropertyKeys) {
+            syncAttributeProfileProperties(claimProperties, allowedClaimProfiles, propertyKey);
+        }
+    }
+
+    /**
+     * Remove the profile properties if the profile property value is the same as the global property value.
+     * If all the profile properties have the same value, change the global property value to the profile value.
+     *
+     * @param claimProperties      Map of claim properties to be updated.
+     * @param allowedClaimProfiles Set of allowed claim profiles.
+     * @param propertyKey          Property key to be updated.
+     */
+    private void syncAttributeProfileProperties(Map<String, String> claimProperties, Set<String> allowedClaimProfiles,
+                                                String propertyKey) {
+
+        String globalValue = claimProperties.get(propertyKey);
+        boolean allProfileValuesMatch = true;
+        String consistentProfileValue = null;
+
+        for (String profileName : allowedClaimProfiles) {
+            String profilePropertyKey = ClaimConstants.PROFILES_CLAIM_PROPERTY_PREFIX + profileName +
+                    ClaimConstants.CLAIM_PROFILE_PROPERTY_DELIMITER + propertyKey;
+            String profilePropertyValue = claimProperties.get(profilePropertyKey);
+
+            // Remove the profile property if it is the same as the global property.
+            if (StringUtils.equals(globalValue, profilePropertyValue)) {
+                claimProperties.remove(profilePropertyKey);
+                allProfileValuesMatch = false;
+                continue;
+            }
+
+            // If a mismatch is already detected, no need to check further for consistency.
+            if (!allProfileValuesMatch) {
+                continue;
+            }
+
+            if (StringUtils.isBlank(profilePropertyValue)) {
+                allProfileValuesMatch = false;
+                continue;
+            }
+            if (consistentProfileValue == null) {
+                consistentProfileValue = profilePropertyValue;
+            } else if (!StringUtils.equals(consistentProfileValue, profilePropertyValue)) {
+                allProfileValuesMatch = false;
+            }
+        }
```

Comment:
Let's try to find a simple algorithm.

// Variable Initialize
consistent_value = value of first item of the list.

// Loop
Iterate over the list.
Check whether match with global.
   - Remove redundant
Check value with consistent_value with current value.
   - Mark as inconsistent if is differ.

Reply:
Updated with clear variable names.